src/node_crypto.cc

```diff
@@ -923,7 +923,20 @@ void SecureContext::SetECDHCurve(const FunctionCallbackInfo<Value>& args) {
 
   node::Utf8Value curve(env->isolate(), args[0]);
 
-  int nid = OBJ_sn2nid(*curve);
+  SSL_CTX_set_options(sc->ctx_, SSL_OP_SINGLE_ECDH_USE);
+
+#if (defined SSL_CTX_set1_curves_list || defined SSL_CTRL_SET_CURVES_LIST)
+# if defined(SSL_CTX_set_ecdh_auto)
+  SSL_CTX_set_ecdh_auto(sc->ctx_, 1);
+# endif
+
+  if (strcmp(*curve, "auto") == 0)
+    return;
+
+  if (!SSL_CTX_set1_curves_list(sc->ctx_, *curve))
+    return env->ThrowError("Failed to set ECDH curve");
+#else
+  int nid = OBJ_sn2nid(strcmp(*curve, "auto") == 0 ? "prime256v1" : *curve);
```

Comment:
I'm not sure we should be implicitly selecting our own hardcoded curve in this case for "auto." We should probably throw an error instead.

Reply:
My intention was to use `tls.DEFAULT_ECDH_CURVE`'s value but displaying error is perhaps better if auto is actually not supported the version of OpenSSL.